proxy/http2/HPACK.cc

```diff
@@ -644,12 +621,11 @@ encode_literal_header_field_with_new_name(uint8_t *buf_start, const uint8_t *buf
 
   // Convert field name to lower case to follow HTTP2 spec.
   // This conversion is needed because WKSs in MIMEFields is old fashioned
-  int name_len;
-  const char *original_name = header.name_get(&name_len);
+  int name_len              = header.name.size();
+  const char *original_name = header.name.data();
 
   ts::LocalBuffer<char> local_buffer(name_len);
```

Comment:
How is `LocalBuffer<char> local_buffer(name_len);` better than `char local_buffer[name_len];`?

Reply:
In case of `name_len` is huge to allocate on thread stack. If `name_len` is small, it's equivalent.